runtime/src/identity.rs

```diff
@@ -211,15 +216,13 @@ decl_module! {
                 ExistenceRequirement::KeepAlive
                 )?;
 
-            // 2.1. Link  master key and signing keys.
+            // 2.1. Link  master key and add pre-authorized signing keys
             Self::link_key_to_did( &master_key, KeyType::External, did);
-            for sig_key in &signing_keys {
-                Self::link_key_to_did( &sig_key.key, sig_key.key_type, did);
-            }
+            signing_keys.iter().for_each( |skey| Self::add_pre_join_identity( skey, did));
```

Comment:
I think this is assuming that `signing_keys` are all type External? If one of them had type `Identity` then it can never be authorised through `authorize_join_to_identity` I believe (since the origin key won't match the Identity Key).